### PR TITLE
SONARIAC-1547 Docker parser should not crash when characters are positioned after EXEC form

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ExecFormImpl.java [hunk 1]
```diff
@@ -45,6 +48,9 @@ public List<Tree> children() {
     result.add(leftBracket);
     result.addAll(argumentsWithSeparators.elementsAndSeparators());
     result.add(rightBracket);
+    if (garbage != null) {
+      result.add(garbage);
```
Comment (petertrr) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ExecFormImpl.java:52]: I wonder if we should add it here. Would this mean that we will highlight this part of the instruction even if it's ignored by both Docker and us?

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java [hunk 2]
```diff
@@ -96,6 +96,8 @@ public class DockerLexicalConstant {
 
   public static final String IMAGE_ALIAS = "[-a-zA-Z0-9_\\.]+";
 
+  public static final String GARBAGE = "[^\n\r]+";
```
Comment (petertrr) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java:99]: That's a funny name :D But I have nothing better to suggest (`IGNORED`? `IGNORED_TRAILING`?)

**Review by petertrr — APPROVED** [review]
LGTM! Please see my suggestion before merging

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ExecFormImpl.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/ExecFormImpl.java:52]: I didn't add it at the beginning, but then it failed other parts of the parser.
The way our parser works requires us to provide in children all elements that leads to/is a `SyntaxToken`.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java (hunk 2 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerLexicalConstant.java:99]: Haha same I couldn't come with something better :sweat_smile: I will try to replace it with something more meaningful.

**Pull request merged**